plumbing/format/diff/patch.go

```diff
@@ -0,0 +1,53 @@
+package diff
+
+import (
+	"io"
+
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/filemode"
+)
+
+type Operation int
+
+const (
+	Equal Operation = iota
+	Add
+	Delete
+)
+
+// Patch represents the necessary steps to transform several files to another.
+type Patch interface {
+	FilePatches() []FilePatch
+	Message() string
+}
+
+// Patcheable is a Patch but with several ways to encode it. It is used to be
+// able to use Patch in a fluent way.
+type Patcheable interface {
```

Comment:
I rewrote the documentation, maybe is clear now. If not, can you suggest a better name for this interface?